File: tests/Generator/resource/typescript/complex/TypeSchema.ts
import type {DefinitionType} from "./DefinitionType";
import type {ArrayDefinitionType} from "./ArrayDefinitionType";
import type {MapDefinitionType} from "./MapDefinitionType";
import type {StructDefinitionType} from "./StructDefinitionType";

/**
 * TypeSchema specification
 */
export interface TypeSchema {
    definitions?: Record<string, ArrayDefinitionType|MapDefinitionType|StructDefinitionType>
    import?: Record<string, string>
    root?: string
}

File: tests/Generator/resource/python/complex/type_schema.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .definition_type import DefinitionType
from .array_definition_type import ArrayDefinitionType
from .map_definition_type import MapDefinitionType
from .struct_definition_type import StructDefinitionType


# TypeSchema specification
class TypeSchema(BaseModel):
    definitions: Dict[str, Annotated[Union[ArrayDefinitionType, MapDefinitionType, StructDefinitionType], Field(discriminator="type")]] = Field(alias="definitions")
    import_: Optional[Dict[str, str]] = Field(default=None, alias="import")
    root: Optional[str] = Field(default=None, alias="root")
    pass


